docs-client/src/containers/MethodPage/GraphqlRequestBody.tsx

```diff
@@ -0,0 +1,190 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useCallback, useEffect, useReducer, useState } from 'react';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import loadable from '@loadable/component';
+
+import {
+  buildClientSchema,
+  GraphQLSchema,
+  getIntrospectionQuery,
+} from 'graphql';
+import TextField from '@material-ui/core/TextField';
+import jsonPrettify from '../../lib/json-prettify';
+import { docServiceDebug } from '../../lib/header-provider';
+
+const graphqlPlaceHolder = `{
+  field(arg: "value") {
+    subField
+  }
+}`;
+const jsonPlaceHolder = jsonPrettify('{"foo":"bar"}');
+
+const QueryEditor = loadable(() => import('./QueryEditor'));
+
+interface Props {
+  requestBodyOpen: boolean;
+  requestBody: string;
+  onEditRequestBodyClick: React.Dispatch<unknown>;
+  onDebugFormChange: (value: string) => void;
+  schemaUrlPath: string;
+}
+
+const toggle = (prev: boolean, override: unknown) => {
+  if (typeof override === 'boolean') {
+    return override;
+  }
+  return !prev;
+};
+
+const parseJson = (requestBody: string) => {
+  let parsedJson;
+  try {
+    parsedJson = JSON.parse(requestBody);
+  } catch (e) {
+    // ignored
+  }
+  return parsedJson;
+};
+
+const GraphqlRequestBody: React.FunctionComponent<Props> = ({
+  requestBodyOpen,
+  requestBody,
+  onEditRequestBodyClick,
+  onDebugFormChange,
+  schemaUrlPath,
+}) => {
+  const [queryOpen, toggleQueryOpen] = useReducer(toggle, true);
+  const [variablesOpen, toggleVariablesOpen] = useReducer(toggle, false);
+
+  const [query, setQuery] = useState('');
+  const [variables, setVariables] = useState('');
+  const [schema, setSchema] = useState<GraphQLSchema | undefined>();
+
+  useEffect(() => {
+    (async () => {
+      const headers: any = {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      };
+      if (process.env.WEBPACK_DEV === 'true') {
+        headers[docServiceDebug] = 'true';
+      }
+      const httpResponse = await fetch(schemaUrlPath, {
+        method: 'POST',
+        headers,
+
+        body: JSON.stringify({
+          operationName: 'IntrospectionQuery',
+          // See https://github.com/graphql/graphiql/blob/8ac05f8b141b6f5cb4449c62ad67a34115490ac8/packages/graphiql/src/utility/introspectionQueries.ts#L16...L22
+          query: getIntrospectionQuery().replace(
+            'subscriptionType { name }',
+            '',
+          ),
+        }),
+      });
+      const result = await httpResponse.json();
+      if (typeof result !== 'string' && 'data' in result) {
+        setSchema(buildClientSchema(result.data));
+      }
+    })();
+  }, [schemaUrlPath]);
+
+  useEffect(() => {
```

Comment:
Q) May I ask what the purpose of this effect is?

Reply:
It seems to have been written differently than intended. I guess this should be extracted from `useState`.

ex)
```javascript
const [query, setQuery] = useState(extractQuery(requestBody));
...
```

Reply:
I see, why should `query`, `variable` be set based on `requestBody`? I thought they were user-provided values 🤔 

From what I understand, they have a dependency on each other at the moment.. 
update `query`/`variable` -> update `requestBody` -> update `query`/`variable` ...

Reply:
I thought code similar to this already existed. 
See https://github.com/line/armeria/blob/master/docs-client/src/containers/MethodPage/DebugPage.tsx#L148...L194

Then, in the case of graphql service, is it better to extract the `requestBody` from the `DebugPage` component and pass the `query` and `variable` to the `GraphqlRequestBody` component??

Reply:
Ah I didn't realize `requestBody` could be parsed from the url as well 😅  Changes look good to me 👍